src/components/Footer/Footer.js

```diff
@@ -63,23 +87,117 @@ const Footer = ({
     }
     return COLORS.balance_gray;
   };
+  const username = item?.anon_user_info_emoji_name
+    ? `${item.anon_user_info_color_name} ${item?.anon_user_info_emoji_name}`
+    : item?.actor?.data?.username;
+
+  const onPressDM = async () => {
+    try {
+      setLoading({...loading, loadingDm: true});
+      if (!item?.anon_user_info_emoji_name) {
+        const channelName = username;
+        const selectedUser = {
+          user: {
+            name: channelName,
+            image: item?.actor?.data?.profile_pic_url || DEFAULT_PROFILE_PIC_PATH
+          }
+        };
+        const members = [item?.actor?.id, profile.myProfile.user_id];
+        await createSignChat(members, selectedUser);
+      } else {
+        await sendMessageDM(item.id, 'post', 'SIGNED');
+      }
+    } catch (e) {
+      console.warn(e);
+    } finally {
+      refSheet.current.close();
+      setLoading({...loading, loadingDm: false});
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling here is done by logging the error to the console. This might not be very helpful for end users. Consider adding a user-friendly error message or some form of feedback to let the user know something went wrong.



<!-- This is an auto-generated comment by OSS CodeRabbit -->